src/language-switcher/class-wpml-elementor-language-switcher.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace WPML\PB\Elementor\LanguageSwitcher;
+
+class LanguageSwitcher implements \IWPML_Action {
```

Comment:
@amElnagdy You can change this to implement `\IWPML_Backend_Action` and `\IWPML_Frontend_Action` and you wont need the factory.